### PR TITLE
Add FLEA job_error endpoint

### PR DESCRIPTION
**Pull request by tpdocai**
Also, fix unit test and change auth_err to authErr to make linter happier.

### CODE REVIEW

File: e2e/create-sample-tasks.sh [hunk 1]
```diff
@@ -128,3 +128,10 @@ curl -s  -H "Authorization: ClientToken" $FLEA_URL/start_task/b7
 echo ""
 echo "Unauthorized 2:"
 curl -s  $FLEA_URL/start_task/b7
+
+echo
+echo "Task Error Report (fails on memory backend - invalid JobId, will work under GCS backend):"
```
Comment (nkashy1) [inline e2e/create-sample-tasks.sh]: @tpdocai : Once the ICML rush is finished, let's sit down and figure out how to make these tests pass easily on the `memory` backend.

The goal is for us to be able to run end-to-end tests against `memory` backend in our continuous integration environment.

File: storage/gcs/flea.go [hunk 2]
```diff
@@ -180,3 +184,24 @@ func (store flea) ListTasks(ctx context.Context, req api.ListTasksRequest) (api.
...
+	_, err := store.GetTask(ctx, req.TaskId)
+	if err != nil {
+		return err
+	}
+
+	if !common.IsValidID(req.JobId) {
+		return storage.ErrInvalidJobId
+	}
+
+	// We only store the last error.
```
Comment (nkashy1) [inline storage/gcs/flea.go]: Not sure if this is the right behaviour for the client, but let's revisit it after this PR -- we should make sure to communicate this behaviour to @phildow .

File: e2e/create-sample-tasks.sh [hunk 3]
```diff
@@ -128,3 +128,10 @@ curl -s  -H "Authorization: ClientToken" $FLEA_URL/start_task/b7
...
+curl -s -X POST -H "Authorization: Bearer ClientToken" \
```
Comment (nkashy1) [inline e2e/create-sample-tasks.sh]: This `curl` request generates the following error:
```
{"error":"json: cannot unmarshal number into Go struct field TaskDetails.deadline of type timestamp.Timestamp","message":"json: cannot unmarshal number into Go struct field TaskDetails.deadline of type timestamp.Timestamp","code":2,"details":[]}
```

File: e2e/create-sample-tasks.sh (hunk 1 above)
Comment (tpdocai) [reply e2e/create-sample-tasks.sh]: The test does not fail. It just returns job-ID not found. See last line in expectations file. The diff expected output test works against memory backend.

File: storage/gcs/flea.go (hunk 2 above)
Comment (tpdocai) [reply storage/gcs/flea.go]: The right behavior is to put it in SQL database. This is cludgy, but we don't even store the job IDs.

File: e2e/create-sample-tasks.sh (hunk 3 above)
Comment (tpdocai) [reply e2e/create-sample-tasks.sh]: This is NOT what I get. It is possible that a call with Deadline as number somehow made it through and that's a response to that. Needs some debugging.

**Comment by tpdocai** [comment]
Re: make failing... I assume that if you have GOOGLE_ACCESS_ID and PRIVATE_PEM_KEY defined, you want to test the URL signing, which requires you to provide FLEA_GCS_UPLOAD_BUCKET. The fix is to undefine one of these variables or to define them all.

File: e2e/create-sample-tasks.sh (hunk 3 above)
Comment (tpdocai) [reply e2e/create-sample-tasks.sh]: I tested providing deadline in the wrong format and it gets rejected prior to reaching my code:
 curl -s -X POST -H "Authorization: Bearer TaskGenToken"   -H "Content-Type: application/json"  -d "{\"deadline\": \"1001049\", \"active\": false}" localhost:8083/v1/flea/modify_task/x3
{"error":"bad Timestamp: parsing time \"1001049\" as \"2006-01-02T15:04:05.999999999Z07:00\": cannot parse \"049\" as \"-\"","message":"bad Timestamp: parsing time \"1001049\" as \"2006-01-02T15:04:05.999999999Z07:00\": cannot parse \"049\" as \"-\"","code":3,"details":[]}

Could it be that your error comes due to invalid/old data on GCS maybe?

Comment (nkashy1) [reply e2e/create-sample-tasks.sh]: @tpdocai : I was testing against the GCS backend (tasks on `tensorio-models-backend-dev` bucket). I blew out the `tasks/` directory and re-ran `e2e/create-sample-tasks.sh`. It worked this time. It seems as if the deadline formatting was indeed the issue.

File: e2e/create-sample-tasks.sh (hunk 1 above)
Comment (nkashy1) [reply e2e/create-sample-tasks.sh]: @tpdocai : The fix I recommend (not a blocker on this PR) is to:

1. Create a job ID by subscribing to a job in the `create-sample-tasks.sh` script (this is already being done)

1. Extract the `jobId` from the response using `jq`.

1. Use that extract jobId to report the error.

This should fix the problem on memory backend.

File: storage/gcs/flea.go (hunk 2 above)
Comment (nkashy1) [reply storage/gcs/flea.go]: In that case, we should prioritize switching over to a SQL database backend. It is useful to track job subscriptions for a task. What do you think?

File: e2e/create-sample-tasks.sh (hunk 1 above)
Comment (tpdocai) [reply e2e/create-sample-tasks.sh]: Good point. Fixed.

File: storage/gcs/flea.go (hunk 2 above)
Comment (tpdocai) [reply storage/gcs/flea.go]: We presently don't have APIs that query the jobs, so it that's even a NOOP, but yes, we should switch to SQL and reimplement.

**Pull request merged**